src/app/public/modules/data-manager/data-manager.service.ts

```diff
@@ -284,4 +302,25 @@ export class SkyDataManagerService implements OnDestroy {
 
     this.viewkeeperClasses.next(viewkeeperClasses);
   }
+
+  private filterDataStateProperties(state: SkyDataManagerState, properties: string[]): SkyDataManagerState {
+    const stateProperties = state.getStateOptions() as { [key: string]: any; };
+    let filteredStateProperties: any = {};
+    for (let property of properties) {
+      /* istanbul ignore else */
+      if (stateProperties.hasOwnProperty(property)) {
+        filteredStateProperties[property] = stateProperties[property];
+      }
+    }
+
+    return new SkyDataManagerState(filteredStateProperties);
```

Comment:
Do you need to create a new `SkyDataManagerState` object here? Could you return the raw `filteredStateProperties` and then compare that instead?